frontend/src/features/settings/pages/Suspend.tsx

```diff
@@ -0,0 +1,139 @@
+import { useContext, useEffect, useState } from "react";
+import { Switch } from "@mui/material";
+
+import "./Suspend.scss";
+import { SuspendModal } from "../components/suspend/SuspendModal";
+import { useSuspendCompanyMutation, useSuspensionHistoryQuery } from "../hooks/suspend";
+import { getDefaultRequiredVal } from "../../../common/helpers/util";
+import { SuspensionHistory } from "../components/suspend/SuspensionHistory";
+import { SUSPEND_ACTIVITY_TYPES } from "../types/suspend";
+import OnRouteBCContext from "../../../common/authentication/OnRouteBCContext";
+import { canUpdateSuspend } from "../helpers/permissions";
+
+export const Suspend = ({
+  companyId,
+  onHideTab,
+}: {
+  companyId: number;
+  onHideTab?: (hide: boolean) => void;
+}) => {
+  const {
+    data: suspensionHistory,
+    isError: fetchHistoryError,
+    refetch: refetchSuspensionHistory,
+  } = useSuspensionHistoryQuery(companyId);
+
+  // Check if user can update suspend
+  const {
+    userRoles,
+    isCompanySuspended,
+    setIsCompanySuspended,
+  } = useContext(OnRouteBCContext);
+
+  const canSuspendCompany = canUpdateSuspend(userRoles);
+
+  const suspendCompanyMutation = useSuspendCompanyMutation();
+
+  const suspensionHistoryList = fetchHistoryError ?
+    [] : getDefaultRequiredVal([], suspensionHistory);
+  
+  const [showSuspendModal, setShowSuspendModal] = useState<boolean>(false);
+
+  // Let Settings dashboard hide "Suspend" tab if:
+  // User isn't allowed to suspend company AND company has never been suspended before
+  useEffect(() => {
+    onHideTab?.(
+      !canSuspendCompany
+      && (suspensionHistory != null)
+      && (suspensionHistory.length === 0)
+    );
+  }, [canSuspendCompany, suspensionHistory]);
+
+  useEffect(() => {
+    // Get current suspension status for company from the most recent entry in suspension history
+    // If suspension history is empty, it's assumed that the company has not yet been suspended
+    // This should be used as the source of truth for determining if the company is currently suspended.
+    const isCurrentlySuspended = suspensionHistoryList.length > 0
+      ? suspensionHistoryList[0].suspendActivityType === SUSPEND_ACTIVITY_TYPES.SUSPEND_COMPANY
+      : false;
+
+    setIsCompanySuspended?.(isCurrentlySuspended);
+  }, [suspensionHistoryList]);
+
+  const isActionSuccessful = (status: number) => {
+    return status === 200 || status === 201;
+  };
+
+  const handleSuspendCompany = async (reason: string) => {
+    const suspendResult = await suspendCompanyMutation.mutateAsync({
+      companyId,
+      data: {
+        suspendActivityType: SUSPEND_ACTIVITY_TYPES.SUSPEND_COMPANY,
+        comment: reason,
+      },
+    });
+
+    if (isActionSuccessful(suspendResult.status)) {
+      refetchSuspensionHistory();
+      setShowSuspendModal(false);
```

Comment:
Isn't this a double call because the useSuspendMutation hook already takes care of this? applicable for the following unsuspend operation too

Reply:
The double call as in invalidating/refetching the suspension history? One issue I ran into (not just for suspend but also for creating applications), was that after successful action, the invalidateQueries inside the mutations doesn't work, and we need to call setState/refetch method from inside the actual page (eg. handleSubmit or whatever)... Should I remove the invalidateQueries instead?

Reply:
You could do one of the following
a) remove it from the hook and keep everything contained here.
b) In the hook, instead of async await, just use a normal function and use an onSuccess function and invalidate queries there [Refer](https://github.com/bcgov/onroutebc/blob/main/frontend/src/features/manageVehicles/apiManager/hooks.ts#L65)

Reply:
I'll try a), remove the invalidateQueries from the mutation hook. I actually tried b) before but it still had issues.